yoticlient.go

```diff
@@ -222,26 +220,21 @@ func getProtobufAttribute(profile Profile, key string) *yotiprotoattr.Attribute
 	return nil
 }
 
-func handleSuccessfulResponse(responseContent string, key *rsa.PrivateKey) (userProfile UserProfile, activityDetails ActivityDetails, errStrings []string) {
+func handleSuccessfulResponse(responseContent string, key *rsa.PrivateKey) (userProfile UserProfile, activityDetails ActivityDetails, err error) {
 	var parsedResponse = profileDO{}
-	var err error
 
 	if err = json.Unmarshal([]byte(responseContent), &parsedResponse); err != nil {
-		errStrings = append(errStrings, err.Error())
 		return
 	}
 
 	if parsedResponse.Receipt.SharingOutcome != "SUCCESS" {
 		err = ErrSharingFailure
-		errStrings = append(errStrings, err.Error())
 	} else {
 		var attributeList, appAttributeList *yotiprotoattr.AttributeList
 		if attributeList, err = decryptCurrentUserReceipt(&parsedResponse.Receipt, key); err != nil {
-			errStrings = append(errStrings, err.Error())
 			return
```

Comment:
The previous assignment happened on the other branch of an if statement, so errStrings was null here... then we immediately return so nothing else can be appended. So errStrings can only be length 1